src/js/fs.js

```diff
@@ -382,6 +382,182 @@ fs.readdirSync = function(path) {
 };
 
 
+try {
+  var stream = require('stream');
+  var Readable = stream.Readable;
+  var Writable = stream.Writable;
+
+  // Throw error if the stream module does not exist
+  // FIXME: require should throw an error when requiring a non-existent module
+  if (stream === {}) {
+    throw new Error('stream is not supported in the current build');
+  }
+
+  function ReadStream(path, options) {
+    if (!(this instanceof ReadStream)) {
+      return new ReadStream(path, options);
+    }
+
+    var defaultOptions = {
+      flags: 'r',
+      encoding: null,
+      fd: null,
+      mode: 438, // == octal 0666
+      autoClose: true,
+      bufferSize: 4096,
+    };
+
+    options = options || defaultOptions;
```

Comment:
What will happen if someone only specify one option? For example:
`fs.createReadStream("ble.txt", {flags: 'r'})`